kernel/timeout.c

```diff
@@ -83,8 +83,18 @@ static s32_t next_timeout(void)
 	return ret;
 }
 
-void z_add_timeout(struct _timeout *to, _timeout_func_t fn, s32_t ticks)
+void z_add_timeout(struct _timeout *to, _timeout_func_t fn, k_timeout_t t)
```

Comment:
could it take `k_ticks_t` as argument?

Reply:
Yes, but why?  In the new design, it just passes through a k_timeout_t that was constructed all the way up in application code without having to convert or inspect anything in the intervening layers.  That's better, I'd argue.